container_pipeline/workers/scanners/runner.py

```diff
@@ -0,0 +1,192 @@
+            ScannerRPMVerify,
+            "registry.centos.org/pipeline-images/misc-package-updates":
+            MiscPackageUpdates,
+            "registry.centos.org/pipeline-images/"
+            "container-capabilities-scanner":
+            ContainerCapabilities
+        }
+
+
+    def pull_image_under_test(self, image_under_test):
+        """
+        Pulls image under test on test run host
+        """
+        self.logger.info("Pulling image %s" % image_under_test)
+        pull_data = conn.pull(
+            repository=image_under_test
+        )
+
+        if 'error' in pull_data:
+            self.logger.fatal("Error pulling requested image {}: {}".format(
+                image_under_test, pull_data
+            ))
+            return False
+        self.logger.info("Image is pulled %s" % image_under_test)
+        return True
+
+    def run_a_scanner(self, scanner, image_under_test):
+        """
+        Run the given scanner on image_under_test
+        """
+        json_data = {}
+        # create object for the respective scanner class
+        scanner_obj = self.scanners.get(scanner)()
+        # should receive the JSON data loaded
+        status, json_data = scanner_obj.scan(image_under_test)
+
+        if not status:
+            self.logger.warning("Failed to run the scanner %s" % scanner)
+        else:
+            self.logger.info("Finished running %s scanner." % scanner)
+
+        # if scanner failed to run, this will return {}, do check at receiver
+        # end
+        return json_data
+
+    def export_scanners_status(self, status, status_file_path):
+        """
+        Export status of scanners execution for build in process
+        """
+        try:
+            fin = open(status_file_path, "w")
+            json.dump(status, fin)
+        except IOError as e:
+            self.logger.critical("Failed to write scanners status on NFS share.")
+            self.logger.critical(str(e))
+        else:
+            self.logger.info(
+                "Wrote scanners status to file: %s" % status_file_path)
+
+    def export_scanner_logs(self, scanner, data):
+        """
+        Export scanner logs in given directory
+        """
+        logs_file_path = os.path.join(
+                self.job_info["logs_dir"],
+                constants.SCANNERS_RESULTFILE[scanner][0])
+        self.logger.info(
+                "Scanner=%s result log file:%s" % (scanner, logs_file_path)
+        )
+        try:
+            fin = open(logs_file_path, "w")
+            json.dump(data, fin, indent=4, sort_keys=True)
+        except IOError as e:
+            self.logger.critical(
+                "Failed to write scanner=%s logs on NFS share." % scanner)
+            self.logger.critical(str(e), exc_info=True)
+        else:
+            self.logger.info(
+                "Wrote the scanner logs to log file: %s" % logs_file_path)
+        finally:
+            return logs_file_path
```

Comment:
So, irrespective of whether there was an exception or not, we are returning the `logs_file_path` variable. The place where this is being called, there's a comment which goes like:
```python
# TODO: what to do if the logs writing failed, check status here
# scanners results file path on NFS
logs_filepath = self.export_scanner_logs(scanner, data_temp)
```
Again, it's an unaddressed `TODO`.